src/crypto.js

```diff
@@ -9,6 +10,21 @@ const toBuf = (doWork, other) => (input) => {
   return new Buffer(doWork(input, other), 'hex')
 }
 
+const fromString = (doWork, other) => (_input) => {
+  const input = Buffer.isBuffer(_input) ? _input.toString() : _input
+  return doWork(input, other)
+}
+
+const to32BitBuf = (doWork, other) => (input) => {
+  let number = doWork(input, other)
+  const bytes = []
+  for (let i = 0; i < 4; i++) {
+    bytes.push(number & 0xff)
+    number = number >>> 8
+  }
```

Comment:
Could you test if this gets faster if you unroll this loop, given these are only 4 iterations the code amount wouldn't be too bad.